inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -425,11 +426,13 @@ def refextract(obj, eng):
         if tmp_document:
             pdf_references = extract_references_from_pdf(tmp_document, source)
             pdf_references = [match_reference(ref) for ref in pdf_references]
+            pdf_references = dedupe_list(pdf_references)
 
     text = get_value(obj.extra_data, 'formdata.references')
     if text:
         text_references = extract_references_from_text(text, source)
         text_references = [match_reference(ref) for ref in text_references]
+        text_references = dedupe_list(text_references)
```

Comment:
This is not tested, if I read correctly.

Reply:
not yet yes, I'm adding a test for it.